### PR TITLE
[ace] Modifications to generate the correct build files for x64-linux configuration

### PR DESCRIPTION
**Pull request by pierrebizz**
Fixes x64-linux configuration not generating correct make files.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
 yes

### CODE REVIEW

**Comment by pierrebizz** [comment]
Fixes https://github.com/microsoft/vcpkg/issues/21525

File: ports/ace/portfile.cmake [hunk 1]
```diff
@@ -104,6 +100,9 @@ if(VCPKG_TARGET_IS_WINDOWS)
   else()
     set(SOLUTION_TYPE vs2022)
   endif()
+	if (VCPKG_LIBRARY_LINKAGE STREQUAL static)
+	  set(MPC_STATIC_FLAG -static)
+	endif()
```
Comment (JackBoosY) [inline ports/ace/portfile.cmake:105]:
```suggestion
  if (VCPKG_LIBRARY_LINKAGE STREQUAL static)
    set(MPC_STATIC_FLAG -static)
  endif()
```

File: ports/ace/portfile.cmake [hunk 2]
```diff
@@ -329,6 +328,10 @@ elseif(VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX)
     list(APPEND _ace_makefile_macros "ssl=1")
   endif()
 
+	if (VCPKG_LIBRARY_LINKAGE STREQUAL static)
+	  list(APPEND _ace_makefile_macros "static_libs_only=1")
+	endif()
```
Comment (JackBoosY) [inline ports/ace/portfile.cmake:333]:
```suggestion
  if (VCPKG_LIBRARY_LINKAGE STREQUAL static)
    list(APPEND _ace_makefile_macros "static_libs_only=1")
  endif()
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

File: ports/ace/portfile.cmake [hunk 3]
```diff
@@ -104,6 +100,9 @@ if(VCPKG_TARGET_IS_WINDOWS)
   else()
     set(SOLUTION_TYPE vs2022)
   endif()
+  if (VCPKG_LIBRARY_LINKAGE STREQUAL static)
```
Comment (strega-nil-ms) [inline ports/ace/portfile.cmake:103]:
```suggestion
  if (VCPKG_LIBRARY_LINKAGE STREQUAL "static")
```

File: ports/ace/portfile.cmake [hunk 4]
```diff
@@ -329,6 +328,10 @@ elseif(VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX)
     list(APPEND _ace_makefile_macros "ssl=1")
   endif()
 
+  if (VCPKG_LIBRARY_LINKAGE STREQUAL static)
```
Comment (strega-nil-ms) [inline ports/ace/portfile.cmake:331]:
```suggestion
  if (VCPKG_LIBRARY_LINKAGE STREQUAL "static")
```

**Review by strega-nil-ms — COMMENTED** [review]
minor changes, then I'll merge post-green

**Comment by strega-nil-ms** [comment]
Thanks!

**Pull request merged**